test/test_warning.rb

```diff
@@ -46,6 +46,90 @@ def teardown
     Warning.clear
   end
 
+  def test_warning_clear_ignore
+    Warning.ignore(/.*/)
+
+    assert_warning '' do
+      Warning.warn 'foo'
```

Comment:
I had to use `Warning.warn` because otherwise tests for Ruby 2.4 https://github.com/jeremyevans/ruby-warning/runs/7250088099?check_suite_focus=true would fail.

It's also worth noting that `warn "foo"` returns a newline while `Warning.warn "foo"` does not so no need to assert `"foo\n"`.

I guess this is not related to this `warning` gem but the Ruby's `Warning` module:

```ruby
$ irb
>> warn "foo"
foo
=> nil
>> Warning.warn "foo"
foo=> nil
```